src/mono/mono/utils/mono-mmap-windows.c

```diff
@@ -123,19 +123,11 @@ mono_valloc_aligned (size_t length, size_t alignment, int flags, MonoMemAccountT
 int
 mono_vfree (void *addr, size_t length, MonoMemAccountType type)
 {
+	int res;
 	MEMORY_BASIC_INFORMATION mbi;
-	SIZE_T query_result = VirtualQuery (addr, &mbi, sizeof (mbi));
-	BOOL res;
-
-	g_assert (query_result);
-
-	res = VirtualFree (mbi.AllocationBase, 0, MEM_RELEASE);
-
-	g_assert (res);
-
+	res = (VirtualQuery (addr, &mbi, sizeof (mbi)) != 0 && VirtualFree (mbi.AllocationBase, 0, MEM_RELEASE) != 0) ? 0 : -1;
```

Comment:
Might just be personal preference but I would find something like this easier to read:

```c
  gboolean success = FALSE;
  if (VirtualQuery (addr, &mbi, sizeof(mbi)) != 0) {
    success = VirtualFree (mbi.AllocationBase, 0, MEM_RELEASE) != 0;
  };
  mono_account_mem (type, -(ssize_t)length);
  return success ? 0 : -1;
```